lib/Pipedrive.js

```diff
@@ -28,6 +30,71 @@
 			callback(error, collectionItems, additionalData);
 		}, false);
 	};
+	exports.OAuthAuthenticate = function (client_id, redirect_uri, res) {
+	    res.redirect('https://oauth.pipedrive.com/oauth/authorize?' +
+            querystring.stringify({
+                client_id: client_id,
+                redirect_uri: redirect_uri,
+            }))
+	};
+
+	exports.OAuthAuthenticateCallback = function (req, res, redirect_uri, clientId, clientSecret) {
+	    var code = req.query.code || null;
+	    var authOptions = {
+	        url: 'https://oauth.pipedrive.com/oauth/token',
+            form: {
+	            code: code,
+                redirect_uri: redirect_uri,
+                grant_type: 'authorization_code'
+            },
+            headers: {
+	            'Authorization': 'Basic ' + (new Buffer(
+	                clientId + ':' + clientSecret
+                ).toString('base64'))
+            },
+            json: true
+	    };
+	    return new Promise(function (resolve, reject) {
+	        request.post(authOptions, function (error, response, body) {
+	            if (error) {
+	                log('OAuthAuthenticateCallback.error handling: ' + error);
+	                reject(error)
+	            }
+	            var accessToken = body.access_token
+                log('access_token: ' + accessToken);
+	            resolve(body)
+	        })
+	    })
+	};
+
+	exports.OAuthRefreshToken = function (req, res, refresh_token, clientId, clientSecret) {
+	    var refreshOptions = {
+	        url: 'https://oauth.pipedrive.com/oauth/token',
+            form: {
+                refresh_token: refresh_token,
+                grant_type: 'refresh_token'
+            },
+            headers: {
+	            'Authorization': 'Basic ' + (new Buffer(
+	                clientId + ':' + clientSecret
+                ).toString('base64'))
+            },
```

Comment:
similar header-building part is in OAuthAuthenticateCallback, maybe this part could be extracted & reused?